### PR TITLE
Nickhand use pytest

### PR DESCRIPTION
**Pull request by rainwoodman**

### CODE REVIEW

**Comment by rainwoodman** [comment]
The basic idea is to move the tests out side of mpi4py_test directory.

The symlink change is necessary to get python2 working. But I think it may interfere with your setup of coverage in nbodykit?

In principle the tests shall not depend on files in the source code directory. Running in coverage mode, we can take an argument which is the path of the configuration file -- this way it is no longer implied in the source directory.

**Comment by nickhand** [comment]
The better thing to do re: symlink is just scan the specified coverage config file for any existing paths and update them to add the path to the build directory I think

**Comment by nickhand** [comment]
Actually, it looks like you can omit modules using a ** syntax which avoids all of these issues. Everything is working nicely for nbodykit....looks ready to merge from my end (maybe after removing that symlink all together)

File: mpi4py_test/__init__.py [hunk 1]
```diff
@@ -241,9 +241,13 @@ def test(config):
...
                 pass
             
+            # XXX: Yu: This is strange.
+            # Tests from the installation shall not depend on any files in the source
+            # code directory. I am disabling it to see if it helps python 2.
+            #
             # set up a symlink to the site dir from the test env
             # this ensures relative paths from the test dir in any configuration files will be correct
-            os.symlink(os.path.join(site_dir, self.PROJECT_MODULE), os.path.join(test_dir, self.PROJECT_MODULE))
+            # os.symlink(os.path.join(site_dir, self.PROJECT_MODULE), os.path.join(test_dir, self.PROJECT_MODULE))
```
Comment (rainwoodman) [inline mpi4py_test/__init__.py]: Well, the symlink crashes python 2 so we have to do something about it...

Comment (nickhand) [inline mpi4py_test/__init__.py]: ah yes, I didn't see that the symlink was the reason...

**Comment by nickhand** [comment]
you can exclude lines from being included in coverage statistics in a `coveragerc` file as

```
[run]
omit = 
    **/test_*
    **/extern/**
```

which  uses the `**`` to match any path leading up to ``extern/`` for example. In the past, I had been doing:

```
[run]
omit = 
    nbodykit/tests/test_*
    nbodykit/extern/*
```

which fails to work because the code is matching patterns against `build/testenv/.../site-packages/nbodykit`

**Comment by rainwoodman** [comment]
The remaining issue is whether to rename, and whether to use this for the non-mpi packages as well.

**Comment by rainwoodman** [comment]
merged as #3

**Pull request closed without merging**